pkg/dns/dns.go

```diff
@@ -70,9 +71,16 @@ func newHandler(IPv6 bool, hosts map[string]string, upstreamServers []string) (d
 	var cc *dns.ClientConfig
 	var err error
 	if len(upstreamServers) == 0 {
-		cc, err = dns.ClientConfigFromFile("/etc/resolv.conf")
-		if err != nil {
-			logrus.WithError(err).Warnf("failed to detect system DNS, falling back to %v", defaultFallbackIPs)
+		if runtime.GOOS != "windows" {
+			cc, err = dns.ClientConfigFromFile("/etc/resolv.conf")
+			if err != nil {
+				logrus.WithError(err).Warnf("failed to detect system DNS, falling back to %v", defaultFallbackIPs)
+				cc, err = newStaticClientConfig(defaultFallbackIPs)
+				if err != nil {
+					return nil, err
+				}
+			}
+		} else {
 			cc, err = newStaticClientConfig(defaultFallbackIPs)
```

Comment:
Should probably add a comment saying that on Windows, we only ever use hard-coded servers for fallback.